deepfence_server/reporters/scan/scan_reporters.go

```diff
@@ -697,7 +697,11 @@ func GetScanResultDiff[T any](ctx context.Context, scan_type utils.Neo4jScanType
 	MATCH (n:` + string(scan_type) + `{node_id: $base_scan_id}) -[r:DETECTED]-> (d)
 	WHERE NOT EXISTS {MATCH (m:` + string(scan_type) + `{node_id: $compare_to_scan_id}) -[:DETECTED]-> (d)}
 	OPTIONAL MATCH (d) -[:IS]-> (e)
-	WITH apoc.map.merge( e{.*}, d{.*, masked: coalesce(d.masked or r.masked or e.masked, false), 
+	OPTIONAL MATCH (s) -[:SCANNED]-> (f)
+	OPTIONAL MATCH (c:ContainerImage{node_id: f.docker_image_id}) -[:HAS] ->(t)
+	OPTIONAL MATCH (t) -[ma:MASKED]-> (ims{node_id:d.node_id})
+	WITH apoc.map.merge( e{.*}, 
+	d{.*, masked: coalesce(d.masked or r.masked or e.masked or head(collect(ma.masked)), false), 
```

Comment:
This can be further optimized as:
```suggestion
	OPTIONAL MATCH (s) -[:SCANNED]-> (f:ContainerImage) -[:HAS]-> (t) -[ma:MASKED]-> (d)
	WITH apoc.map.merge( e{.*}, 
	d{.*, masked: coalesce(d.masked or r.masked or e.masked or head(collect(ma.masked)), false), 
```

Reply:
Done.